tox.ini

```diff
@@ -0,0 +1,68 @@
+[tox]
+envlist = py{35,36,37}-{pytest,setup}{,-all},docs,dev,flake8
+skipsdist = True
+requires = tox-venv
+           setuptools >= 30.0.0
+
+[testenv]
+usedevelop = True
+whitelist_externals = which
+passenv = HOME LC_ALL LC_CTYPE
+description =
+    setup: run tests with python setup.py
+    pytest: run tests with pytest
+deps =
+    pytest-cov
+    !all: pytest-astropy
+    !all: pytest-xdist
+    all: -rpip-requirements-dev
+conda_deps=
+    Cython
+    jinja2
+    numpy
+commands =
+    python --version
+    which python
+    pip freeze -l
+    pytest: pytest {posargs}
+    setup: python setup.py test {posargs}
```

Comment:
I mention this in the comments, but I don't think that it makes sense to run `setup.py test` using `tox`. Basically this means that Astropy is creating another test environment within a `tox` environment, which, in my opinion, is exactly what we'd like to avoid. Maybe in the short term we want to maintain this as a sanity check, but our eventual goal should be to get rid of it.

Reply:
`setup.py test` is even actively discouraged in the docs: https://tox.readthedocs.io/en/latest/example/basic.html#integration-with-setup-py-test-command